src/libraries/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -138,61 +138,39 @@ private void PrepareContent()
 
         private sealed class ReadOnlyStream : DelegatingStream
         {
-            public override bool CanWrite
+            public ReadOnlyStream(Stream innerStream) : base(innerStream)
             {
-                get { return false; }
             }
 
-            public override int WriteTimeout
-            {
-                get { throw new NotSupportedException(SR.net_http_content_readonly_stream); }
-                set { throw new NotSupportedException(SR.net_http_content_readonly_stream); }
-            }
```

Comment:
Shouldn't we implement `WriteTimeout` explicitly as well?
As far as I can tell the test is currently using `MemoryStream` as the inner stream, so this ends up in the base `Stream`'s impl which throws the expected `InvalidOperationException`. With an arbitrary `Stream` I assume we want to guard against this being set?

Reply:
If you get the stream from `StreamContent`, you can't write to it. But something else in the process could be if they have access to the stream used to create the content, and setting the `WriteTimeout` change may be unexpected.

That said, I don't think this matters realistically for this stream since there's no reason to set the timeout if all the writes will throw anyway.

Reply:
I don't think it's necessary, but I also don't think it matters either way.  I put it back.